Content.Server/StationRecords/Systems/GeneralStationRecordConsoleSystem.cs

```diff
@@ -68,8 +94,39 @@ private void UpdateUserInterface(EntityUid uid, GeneralStationRecordConsoleCompo
                 stationRecordsComponent);
         }
 
+        GeneralStationRecordConsoleState newState = new(console.ActiveKey, record, listing, console.Filter);
+        SetStateForInterface(uid, newState);
+    }
+
+    private void SetStateForInterface(EntityUid uid, GeneralStationRecordConsoleState newState)
+    {
         _userInterface
-            .GetUiOrNull(uid, GeneralStationRecordConsoleKey.Key)?
-            .SetState(new GeneralStationRecordConsoleState(console.ActiveKey, record, listing));
+            .GetUiOrNull(uid, GeneralStationRecordConsoleKey.Key)
+            ?.SetState(newState);
+    }
+
+    private bool IsSkippedRecord(GeneralStationRecordsFilter filter,
+        GeneralStationRecord someRecord)
+    {
+        bool isFilter = filter.value.Length > 0;
+        string filterLowerCaseValue = "";
+
+        if (isFilter)
+            filterLowerCaseValue = filter.value.ToLower();
+
+        return filter.type switch
+        {
+            GeneralStationRecordFilterType.Name => someRecord.Name != null
+                && isFilter && !someRecord.Name.ToLower().Contains(filterLowerCaseValue),
```

Comment:
you really can just return if it isn't a filter, you don't have to repeat this every single time